parser-typechecker/src/Unison/Codebase/SqliteCodebase.hs

```diff
@@ -610,18 +644,19 @@ sqliteCodebase debugName root = do
               Set.traverse (Cv.referenceid2to1 (getCycleLen "dependentsImpl"))
                 =<< Ops.dependents (Cv.reference1to2 r)
 
-          syncFromDirectory :: MonadIO m => Codebase1.CodebasePath -> SyncMode -> Branch m -> m ()
-          syncFromDirectory srcRoot _syncMode b =
-            flip State.evalStateT emptySyncProgressState $ do
-              srcConn <- unsafeGetConnection (debugName ++ ".sync.src") srcRoot
-              syncInternal syncProgress srcConn conn $ Branch.transform lift b
+          syncFromDirectory :: MonadUnliftIO m => Codebase1.CodebasePath -> SyncMode -> Branch m -> m ()
```

Comment:
These two methods didn't seem to be closing their connections at all before AFAICT.
They don't close them inside syncInternal either.

Reply:
Great find!